fulp_modules/features/antagonists/bloodsuckers/code/bloodsucker/bloodsucker_sol.dm

```diff
@@ -128,23 +128,66 @@
  * - Having less than 10 Brute & Burn Combined while INSIDE of your Coffin while it isnt Sol.
  * - Sol being over, dealt with by /sunlight/process() [bloodsucker_daylight.dm]
 */
+
+/atom/movable/screen/alert/status_effect/torpor
+	name = "Torpor"
+	desc = "You have returned to the precipice of oblivion once more. Through this you shall recover at the expense of being immensely vulnerable."
+	icon = 'fulp_modules/icons/antagonists/bloodsuckers/bloodsucker_status_effects.dmi'
+	icon_state = "torpor"
+	alerttooltipstyle = "alien"
+
+/datum/status_effect/torpor
+	id = "Torpor"
+	status_type = STATUS_EFFECT_UNIQUE
+	duration = -1
+	tick_interval = -1
+	alert_type = /atom/movable/screen/alert/status_effect/torpor
+	/// Our Bloodsucker's antag datum.
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum
+
+/datum/status_effect/torpor/on_apply()
+	. = ..()
+	var/mob/living/carbon/human/user = owner
+	bloodsuckerdatum = IS_BLOODSUCKER(user)
+
+	to_chat(owner, span_notice("You enter the horrible slumber of deathless Torpor. You will heal until you are renewed."))
+	// Force them to go to sleep
+	REMOVE_TRAIT(owner, TRAIT_SLEEPIMMUNE, BLOODSUCKER_TRAIT)
+	// Without this, you'll just keep dying while you recover.
+	owner.add_traits(list(TRAIT_NODEATH, TRAIT_FAKEDEATH, TRAIT_DEATHCOMA, TRAIT_RESISTLOWPRESSURE, TRAIT_RESISTHIGHPRESSURE), BLOODSUCKER_TRAIT)
+	owner.set_timed_status_effect(0 SECONDS, /datum/status_effect/jitter, only_if_higher = TRUE)
+	// Disable ALL Powers
+	bloodsuckerdatum.DisableAllPowers()
+
+/datum/status_effect/torpor/on_remove()
+	owner.grab_ghost()
+	to_chat(owner, span_warning("You have recovered from Torpor."))
+	owner.remove_traits(list(TRAIT_NODEATH, TRAIT_FAKEDEATH, TRAIT_DEATHCOMA, TRAIT_RESISTLOWPRESSURE, TRAIT_RESISTHIGHPRESSURE), BLOODSUCKER_TRAIT)
+	if(!HAS_TRAIT(owner, TRAIT_MASQUERADE))
+		ADD_TRAIT(owner, TRAIT_SLEEPIMMUNE, BLOODSUCKER_TRAIT)
+	bloodsuckerdatum.heal_vampire_organs()
+	SEND_SIGNAL(src, BLOODSUCKER_EXIT_TORPOR)
+
 /datum/antagonist/bloodsucker/proc/check_begin_torpor(SkipChecks = FALSE)
 	var/mob/living/carbon/user = owner.current
+	/// We can't Torpor if already in Torpor
+	if(user.has_status_effect(/datum/status_effect/torpor))
+		return
```

Comment:
why is this called if we're already in torpor?

Reply:
I originally had it like this because I noticed a bug where Bloodsuckers would be spammed with erroneous "you have exited Torpor" messages when at zero blood volume, but since the conversion of Torpor into a status effect that bug has somehow been indirectly corrected again. These lines have now been removed.